shells/pipes-shell/source/verbs/run-arc.js

```diff
@@ -17,35 +17,52 @@ import {portIndustry} from '../pec-port.js';
 const {log, warn} = logsFactory('pipe');
 
 // The implementation was forked from verbs/spawn.js
-export const runArc = async (msg, bus, runtime, env) => {
+export const runArc = async (msg, bus, runtime) => {
   const {recipe, arcId, storageKeyPrefix, pecId, particles} = msg;
-  const action = runtime.context.allRecipes.find(r => r.name === recipe);
   if (!arcId) {
     warn(`arcId must be provided.`);
     return null;
   }
-  if (recipe && !action) {
-    warn(`found no recipes matching [${recipe}]`);
+  const action = await marshalRecipe(recipe, runtime.context);
+  if (!action) {
+    warn(`found no recipe matching [${recipe}]`);
     return null;
   }
+  const arc = await marshalArc(runtime, arcId, storageKeyPrefix, pecId, bus);
+  if (await instantiateRecipe(arc, action, particles || [])) {
+    log(`successfully instantiated ${recipe} in Arc:${arcId}`);
+  }
+  return arc;
+};
+
+const marshalRecipe = async (recipeSpec, context) => {
+  let manifest = context;
+  let recipeName = recipeSpec;
+  if (recipeSpec.includes('|')) {
+    let manifestPath;
+    [manifestPath, recipeName] = recipeSpec.split('|');
```

Comment:
maybe instead of encoding both path and name in the recipe parameter, we could just have an additional optional `manifestPath`?